### PR TITLE
Use niv out of nixpkgs

### PR DESCRIPTION
**Pull request by cprussin**

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -5,10 +5,10 @@
   niv ? sources.niv,
```
Comment (thomashoneyman) [inline shell.nix:5]: Is this still needed if `niv` is being pulled from nixpkgs?

Comment (cprussin) [reply shell.nix:5]: it is not!  great catch, thanks

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ on:
       - "v[0-9]*.[0-9]*.x"
 
 jobs:
-  Setup Environment:
+  Setup_Environment:
```
Comment (davezuch) [inline .github/workflows/check.yaml]: Are step names not allowed to have spaces?

Comment (cprussin) [reply .github/workflows/check.yaml]: unfortunately seems like they are not: https://github.com/citizennet/purescript-httpure/actions/runs/2272722569

> The workflow is not valid. .github/workflows/check.yaml (Line: 11, Col: 3): The identifier 'Setup Environment' is invalid. IDs may only contain alphanumeric characters, '_', and '-'. IDs must start with a letter or '_' and and must be less than 100 characters. .github/workflows/check.yaml (Line: 62, Col: 3): The identifier 'Validate Purescript Format' is invalid. IDs may only contain alphanumeric characters, '_', and '-'. IDs must start with a letter or '_' and and must be less than 100 characters.

Comment (davezuch) [reply .github/workflows/check.yaml]: Ah gotcha, thanks for the explanation!

**Review by davezuch — COMMENTED** [review]
Honestly, I don't know Nix well enough yet to confidently review these changes, but they seem sensible to me.

**Comment by cprussin** [comment]
@davezuch no worries, I'm happy to keep lending a hand on the nix side of things especially, given my passion for nix and personal desire to not see nix stripped out of here :)

**Comment by davezuch** [comment]
@cprussin sounds good! It seems we're wanting to start leaning more on Nix on our team anyway, so I'm glad this repo relies on it.

**Comment by cprussin** [comment]
@davezuch looks like the branch protection rules will need to be updated to merge this, given I changed the job names and so the required jobs no longer exist.

Can you either update them or grant me admin so I can update them myself?

**Comment by davezuch** [comment]
@cprussin done and done!

**Pull request merged**